ads/inabox/inabox-messaging-host.js

```diff
@@ -129,10 +132,9 @@ export class InaboxMessagingHost {
       return false;
     }
 
-    const allowedTypes = adFrame.iframe.dataset['ampAllowed'];
-    // having no whitelist is legacy behavior so assume all types are allowed
-    if (allowedTypes &&
-        allowedTypes.split(/\s*,\s*/).indexOf(request['type']) === -1) {
+    const allowedTypes = adFrame.iframe.dataset['ampAllowed'] || '';
+    if (allowedTypes.split(/\s*,\s*/).concat(READ_ONLY_MESSAGES)
```

Comment:
What if someone wants to allow expand but not the read only ones?

I had thought the semantics you proposed were:

* If `ampAllowed` is absent, only allow read-only messages
* Otherwise, allow exactly `ampAllowed` specifies?

Reply:
Changed to always respect `ampAlllowed` if it's present and not empty.